packages/grafana-ui/src/components/Table/TableNG/Cells/PillCell.tsx

```diff
@@ -49,19 +59,6 @@ interface Pill {
 const SPLIT_RE = /\s*,\s*/;
 const TRANSPARENT = 'rgba(0,0,0,0)';
 
-function createPills(pillValues: unknown[], field: Field, theme: GrafanaTheme2): Pill[] {
-  return pillValues.map((pill, index) => {
-    const bgColor = getPillColor(pill, field, theme);
-    const textColor = colorManipulator.getContrastRatio('#FFFFFF', bgColor) >= 4.5 ? '#FFFFFF' : '#000000';
```

Comment:
turns out we have methods that do this. I've created a memoized one at TableNG that's passed into cell renderers so that the cell can speedily and accurately figure this out.